kernel/src/table_features/column_mapping.rs

```diff
@@ -1,58 +1,79 @@
 //! Code to handle column mapping, including modes and schema transforms
-use std::str::FromStr;
+use super::ReaderFeatures;
+use crate::actions::Protocol;
+use crate::table_properties::TableProperties;
 
 use serde::{Deserialize, Serialize};
-
-use crate::{DeltaResult, Error};
+use strum::EnumString;
 
 /// Modes of column mapping a table can be in
-#[derive(Serialize, Deserialize, Debug, Copy, Clone, PartialEq)]
+#[derive(Debug, Default, EnumString, Serialize, Deserialize, Copy, Clone, PartialEq, Eq)]
+#[strum(serialize_all = "camelCase")]
 #[serde(rename_all = "camelCase")]
```

Comment:
aside: Do these classes still need serde support?

Reply:
unfortunately yea, since it is still a field in `GlobalScanState` (which derives `Serialize`/`Deserialize`)